setup.py

```diff
@@ -4,9 +4,12 @@
 # See LICENSE for details.
 
 import setuptools
+import sysconfig
 from Cython.Build import cythonize
 
-
+macros = None
+if sysconfig.get_config_var("abi_thread") == "t":
```

Comment:
This needs a comment explaining why we need these changes

Reply:
It will help to have a link to https://docs.python.org/3/howto/free-threading-python.html

and, is this the right way to check for thread free ?

In the Python docs I see


> The sysconfig.get_config_var("Py_GIL_DISABLED") configuration variable can be used to determine whether the build supports free threading. If the variable is set to 1, then the build supports free threading. This is the recommended mechanism for decisions related to the build configuration.